File: src/main/java/ludwig/ide/NodeStringConverter.java
package ludwig.ide;

import javafx.util.StringConverter;
import ludwig.model.NamedNode;
import ludwig.model.Node;
import ludwig.utils.NodeUtils;

class NodeStringConverter extends StringConverter<Node> {
    @Override
    public String toString(Node object) {
        return NodeUtils.INSTANCE.signature(object);
    }

    @Override
    public NamedNode fromString(String string) {
        return null;
    }
}

File: src/main/kotlin/ludwig/ide/NodeStringConverter.kt
package ludwig.ide

import javafx.util.StringConverter
import ludwig.model.NamedNode
import ludwig.model.Node
import ludwig.utils.signature

internal class NodeStringConverter : StringConverter<Node>() {
    override fun toString(obj: Node): String {
        return signature(obj)
    }

    override fun fromString(string: String): NamedNode? {
        return null
    }
}
